packages/sysutils/busybox/package.mk

```diff
@@ -30,6 +30,9 @@ PKG_SECTION="system"
 PKG_SHORTDESC="BusyBox: The Swiss Army Knife of Embedded Linux"
 PKG_LONGDESC="BusyBox combines tiny versions of many common UNIX utilities into a single small executable. It provides replacements for most of the utilities you usually find in GNU fileutils, shellutils, etc. The utilities in BusyBox generally have fewer options than their full-featured GNU cousins; however, the options that are included provide the expected functionality and behave very much like their GNU counterparts. BusyBox provides a fairly complete environment for any small or embedded system."
 
+PKG_NEED_UNPACK="$PROJECT_DIR/$PROJECT/busybox $PROJECT_DIR/$PROJECT/initramfs/platform_init"
+[ -n "$DEVICE" ] && PKG_NEED_UNPACK+=" $PROJECT_DIR/$PROJECT/devices/$DEVICE/initramfs/platform_init"
```

Comment:
I think we can do better than this - we should move `busybox/scripts/init` and `busybox/scripts/functions` to `initramfs/scripts`.

The stamp calculation in `scripts/unpack` and `scripts/build` should be changed as follows:
```
-STAMP_DEPENDS="$PKG_DIR $PKG_NEED_UNPACK $PROJECT_DIR/$PROJECT/patches/$PKG_NAME"
-[ -n "$DEVICE" ] && STAMP_DEPENDS="$STAMP_DEPENDS $PROJECT_DIR/$PROJECT/devices/$DEVICE/patches/$PKG_NAME"
+STAMP_DEPENDS="$PKG_DIR $PKG_NEED_UNPACK $PROJECT_DIR/$PROJECT/$PKG_NAME $PROJECT_DIR/$PROJECT/patches/$PKG_NAME"
+[ -n "$DEVICE" ] && STAMP_DEPENDS="$STAMP_DEPENDS $PROJECT_DIR/$PROJECT/devices/$DEVICE/$PKG_NAME $PROJECT_DIR/$PROJECT/devices/$DEVICE/patches/$PKG_NAME"
```
so that project and device config is taken into consideration, and triggers a package rebuild if any config changes. This might result in more cleaning (mismatched stamps) than is necessary, though, so I'm open to suggestion - if you don't add the config directories to the stamp then they'll need to be manually handled by additional PKG_NEED_UNPACK, as you have above (and for Kodi).

Then, `initramfs` would be automatically cleaned if any of the `initramfs` scripts, or `platform_init`, is modified. Since `initramfs` is built by `linux`, it would also be necessary to add `initramfs` to PKG_NEED_UNPACK in `package/linux` so that `linux` is cleaned (and rebuilt) whenever `initramfs` is modified.

Personally I'd rather see this done in a separate PR. Either drop the above change and handle in a follow up PR, or revert it in the follow up PR.

Reply:
I like the idea of adding the package name to the stamp calculation in `scripts/unpack` and `scripts/build` and it will work great for `kodi` and `initramfs` (after your suggested changes) and there are also some other packages that could need some refactoring:
- `plymouth-lite`: include files from project, distro or package `splash`-folder
- `samba`: include `smb.conf` from device/project or distro `config`-folder if it exists
- `xorg-server`: include `xorg.conf` from project `xorg` folder if it exists
- `bcm2835-bootloader`: include files from project or distro `config`-folder (should probably be moved to a `bootloader` folder to follow the other projects)

I will keep this RFC pull-request open as a development branch and create smaller PRs from it to make all changes reviewable.